src/components/Inputs/RadioButtons/RadioButtons.js

```diff
@@ -0,0 +1,293 @@
+
+import useErrorMessage from '../../../hooks/useErrorMessage.js'
+import useIncludes from '../../../hooks/useIncludes.js'
+import useInvalid from '../../../hooks/useInvalid.js'
+import { InputLabel } from '../InputLabel'
+import { Body } from '../../Type/Body.js'
+import { COLORS } from '../../Colors.js'
+import styles from './RadioButtons.module.scss'
+
+const codes = {
+  RETURN: 13,
+  SPACE: 32,
+  END: 35,
+  HOME: 36,
+  LEFT: 37,
+  UP: 38,
+  RIGHT: 39,
+  DOWN: 40,
+}
+
+// Wrapping this way facilitates spying with spyOn
+export const focusHelper = {
+  focus: (elementRef) => {
+    elementRef.current.focus()
+  },
+}
+
+function RadioButton({
+  name,
+  value,
+  checked,
+  tabIndex,
+  required,
+  disabled,
+  label,
+  onClick,
+  onChange,
+  ...rest
+}) {
+  const [, includesKeysOrThrow] = useIncludes(['name', 'label'])
+  includesKeysOrThrow({ name, label })
+  const [includesInvalid] = useInvalid(Object.keys(RadioButton.propTypes))
+  includesInvalid(rest)
+
+  const spanRadio = useRef(null)
+
+  // This allows the user to arrow navigate within radio group preserving focus affordance
+  useEffect(() => {
+    if (checked) {
+      focusHelper.focus(spanRadio)
+    }
+  }, [checked])
+
+  return (
+    <label className={styles.RadioButton}>
+      <span
+        role="radio"
+        ref={spanRadio}
+        value={value}
+        aria-checked={checked}
+        tabIndex={tabIndex}
+      >
+        <input
+          type="radio"
+          name={name}
+          value={value}
+          checked={checked}
+          onClick={onClick}
+          onChange={onChange}
+          required={required}
+          disabled={disabled}
+          data-tid={rest['data-tid']}
+        />
+        <aside />
+      </span>
+      <Body.Regular400 color={COLORS.GRAY_PRIMARY}>{label}</Body.Regular400>
+    </label>
+  )
+}
+
+RadioButton.propTypes = {
+  name: PropTypes.string.isRequired,
+  value: PropTypes.string.isRequired,
+  checked: PropTypes.bool,
+  tabIndex: PropTypes.number,
+  required: PropTypes.bool,
+  disabled: PropTypes.bool,
+  label: PropTypes.node.isRequired,
+  'data-tid': PropTypes.string,
+  onClick: PropTypes.func,
+
+  // These will appear if RadioButtonGroup is used with redux-form:
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func,
+  onDragStart: PropTypes.func,
+  onDrop: PropTypes.func,
+  onFocus: PropTypes.func,
+}
+
+export function RadioButtonGroup({
+  options,
+  onChange,
+  formChangeHandler,
+  allCaps = true,
+  name = `radio-button-group-${uuidv4()}`,
+  initialValue = undefined,
+  currentValue,
+  currentError,
+  formTouched,
+  disabled,
+  validator,
+  required,
+  labelCopy,
+  ...rest
+}) {
+  let initialSelected
+  if (currentValue || typeof currentValue === 'boolean') {
+    initialSelected = currentValue
+  } else if (initialValue || typeof initialValue === 'boolean') {
+    initialSelected = initialValue
+  }
+
+  const [selectedValue, setSelectedValue] = useState(initialSelected)
```

Comment:
I'm assuming initialSelected is ok to be `undefined` here ? The code flow seems to suggest it's possible to not hit any of the `if`s and it's not initialized to anything ?

Reply:
Right, if none of the conditions hit it will be as if `useState()` or explicitly `useState(undefined)`